compiler/rustc_resolve/src/late.rs

```diff
@@ -1425,9 +1425,10 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
                                     } else {
                                         LifetimeUseSet::Many
                                     }),
-                                    LifetimeRibKind::Generics { .. }
-                                    | LifetimeRibKind::ConstGeneric
-                                    | LifetimeRibKind::AnonConst => None,
+                                    LifetimeRibKind::Generics { .. } => None,
+                                    LifetimeRibKind::ConstGeneric | LifetimeRibKind::AnonConst => {
```

Comment:
It can happen, I don't remember how exactly, probably in the simplest case Item -> Generics (define `'a`) -> use `'a`.
I basically started with adding panics everywhere, then removing them until the test suite passes, then trying to construct examples for the remaining cases.

Reply:
Indeed, we need `Item` to break the `find_map` and avoid going into a possibly outer `AnonConst`.